### PR TITLE
Clean up OkHttp configuration

### PR DESCRIPTION
**Pull request by MiSikora**
## Description

Our current `OkHttp` client dependency injection configuration needs some improvement. We don't share the dispatcher between multiple clients, we have unused clients, and navigating the `ServersModule` has become challenging due to the extensive DI configuration. To address these issues, I've made the following changes:

- **Moved interceptor configuration**: All interceptor configurations have been relocated to a separate `InterceptorsModule`.
- **Removed unused clients**: Cleaned up the code by removing clients that are no longer in use.
- **Simplified qualifier annotations**: Consolidated the annotations by replacing `@TranscriptsCache`, `@TranscriptsClientBuilder`, and `@TranscriptsClient` with a single `@Transcripts` annotation.
- **Shared connection pool**: Most clients now share the same connection pool, with the exception of the client used for downloads, which is kept separate to allow it do its own thing.

I would like to provide a custom client for `Coil`, but since it resides in the `:ui` module, I've decided not to address it in this PR. This might be revisited in the future.

## Testing Instructions

Test networking of the app. Using APIs, downloads, caching transcripts, etc.

## Checklist
- [x] ~If this is a user-facing change, I have added an entry in CHANGELOG.md~
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [x] ~I have considered whether it makes sense to add tests for my changes~
- [x] ~All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`~
- [x] ~Any jetpack compose components I added or changed are covered by compose previews~
- [x] ~I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.~
 
#### I have tested any UI changes...

- [ ] with different themes
- [ ] with a landscape orientation
- [ ] with the device set to have a large display and font size
- [ ] for accessibility with TalkBack

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: modules/services/servers/src/main/java/au/com/shiftyjelly/pocketcasts/servers/di/InterceptorModule.kt [hunk 1]
```diff
@@ -1,24 +1,188 @@
...
 
 @InstallIn(SingletonComponent::class)
 @Module
-class InterceptorModule {
-    @Provides
-    @CrashLoggingInterceptor
-    fun provideMonitoringInterceptor(): Interceptor = CrashLoggingOkHttpInterceptorProvider
+object InterceptorModule {
+    private val sixMonths = 180.days.inWholeSeconds
+    private val fiveMinutes = 180.minutes.inWholeSeconds
```
Comment (ashiagr) [inline modules/services/servers/src/main/java/au/com/shiftyjelly/pocketcasts/servers/di/InterceptorModule.kt:29]:
```suggestion
    private val fiveMinutes = 5.minutes.inWholeSeconds
```

File: modules/services/servers/src/main/java/au/com/shiftyjelly/pocketcasts/servers/di/InterceptorModule.kt [hunk 2]
```diff
@@ -1,24 +1,188 @@
...
+            responseBuilder = responseBuilder.header(cacheControlHeader, "public, max-age=$fiveMinutes")
+        }
+        responseBuilder.build()
+    }
+
+    private val cacheControlNotesInterceptor = Interceptor { chain ->
+        val request = chain.request()
+        val originalResponse = chain.proceed(request)
+        originalResponse.newBuilder().header(cacheControlHeader, "public, max-age=$sixMonths").build()
+    }
```
Comment (ashiagr) [inline modules/services/servers/src/main/java/au/com/shiftyjelly/pocketcasts/servers/di/InterceptorModule.kt:59]: `cacheControlNotesInterceptor` doesn't seem to be used anymore. Can we remove it?

**Review by ashiagr — APPROVED** [review]
Thanks for cleaning it up! I left two minor comments, aside from that, everything looks good. 🌟

File: modules/services/servers/src/main/java/au/com/shiftyjelly/pocketcasts/servers/di/InterceptorModule.kt (hunk 2 above)
Comment (MiSikora) [reply modules/services/servers/src/main/java/au/com/shiftyjelly/pocketcasts/servers/di/InterceptorModule.kt:59]: Yes, thanks. It was used in an used client. I'll double check to make sure of it.

**Pull request merged**